include/libp2p/security/noise/handshake.hpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_INCLUDE_LIBP2P_SECURITY_NOISE_HANDSHAKE_HPP
+#define LIBP2P_INCLUDE_LIBP2P_SECURITY_NOISE_HANDSHAKE_HPP
+
+#include <libp2p/common/logger.hpp>
+#include <libp2p/connection/raw_connection.hpp>
+#include <libp2p/crypto/crypto_provider.hpp>
+#include <libp2p/crypto/key_marshaller.hpp>
+#include <libp2p/crypto/x25519_provider.hpp>
+#include <libp2p/outcome/outcome.hpp>
+#include <libp2p/peer/peer_id.hpp>
+#include <libp2p/security/noise/crypto/interfaces.hpp>
+#include <libp2p/security/noise/crypto/state.hpp>
+#include <libp2p/security/noise/handshake_message_marshaller.hpp>
+#include <libp2p/security/noise/insecure_rw.hpp>
+#include <libp2p/security/security_adaptor.hpp>
+
+namespace libp2p::security::noise {
+
+  std::shared_ptr<CipherSuite> defaultCipherSuite();
+
+  class Handshake : public std::enable_shared_from_this<Handshake> {
+   public:
+    explicit Handshake(
```

Comment:
Would it not be better to introduce some kind of a builder? This constructor is monstrous.

Reply:
Constructor has a numerous of arguments - agree. Since that class is not exposed to dependency injector and is constructed only inside the Noise adapter - this is ok.